packages/compass-sidebar/src/components/multiple-connections/active-connections/active-connection-navigation.tsx

```diff
@@ -200,34 +198,65 @@ function ActiveConnectionNavigation({
   );
 }
 
-function mapStateToProps(state: RootState) {
-  const {
-    databases: { filterRegex, filteredDatabases, expandedDbList },
-    instance,
-  } = state;
-  const status = instance?.databasesStatus;
-  const isReady =
-    status !== undefined && !['initial', 'fetching'].includes(status);
-  const defaultExpanded = Boolean(filterRegex);
-  const expanded = Object.fromEntries(
-    (filteredDatabases as any[]).map(({ name }) => [
-      name,
-      expandedDbList[name] ?? defaultExpanded,
-    ])
-  );
-  const isDataLake = instance?.dataLake.isDataLake;
-  const isWritable = instance?.isWritable;
+function mapStateToProps(
+  state: RootState,
+  { activeConnections }: { activeConnections: ConnectionInfo[] }
+): {
+  isReady: boolean;
+  connections: Connection[];
+  expanded: Record<string, Record<string, boolean> | false>;
+} {
+  const connections: Connection[] = [];
+  const expandedResult: Record<string, any> = {};
+
+  for (const connectionInfo of activeConnections) {
+    const connectionId = connectionInfo.id;
+    const instance = state.instance[connectionId];
+    const {
+      filterRegex,
+      filteredDatabases,
+      expandedDbList: initialExpandedDbList,
+    } = state.databases[connectionId] || {};
+
+    const status = instance?.databasesStatus;
+    const isReady =
+      status !== undefined && !['initial', 'fetching'].includes(status);
+    const defaultExpanded = Boolean(filterRegex);
+
+    const expandedDbList = initialExpandedDbList ?? {};
+    const expanded = Object.fromEntries(
+      ((filteredDatabases as any[]) || []).map(({ name }) => [
+        name,
+        expandedDbList[name] ?? defaultExpanded,
+      ])
+    );
+
+    const isDataLake = instance?.dataLake?.isDataLake ?? false;
+    const isWritable = instance?.isWritable ?? false;
+
+    connections.push({
+      isReady,
+      isDataLake,
+      isWritable,
+      name: getConnectionTitle(connectionInfo),
+      connectionInfo,
+      databasesLength: filteredDatabases?.length ?? 0,
+      databasesStatus: status as Connection['databasesStatus'],
+      databases: filteredDatabases ?? [],
+    });
+
+    expandedResult[connectionId] = expanded;
+  }
 
   return {
-    isReady,
-    isDataLake,
-    isWritable,
-    databases: filteredDatabases,
-    expanded,
+    isReady: true,
+    connections,
+    expanded: expandedResult,
```

Comment:
The same concern for this - about doing this inside of component
(can be done as a follow up)